### PR TITLE
main/linux-vanilla: add patch to export now GPL-only fpu functions

### PR DESCRIPTION
**Pull request by Cogitri**
This significantly speeds up ZFS' due to allowing it to use SIMD. Other distros
such as NixOS use the same patch.

Also do some minor cleanups while we're at it.

### CODE REVIEW

**Comment by nmeum** [comment]
`kernel_fpu_begin` is `EXPORT_SYMBOL_GPL` but the patch exports `__kernel_fpu_begin` instead of changing the export of `kernel_fpu_begin` to `EXPORT_SYMBOL` does this work around the license compatibility issues outlined in https://lkml.org/lkml/2019/5/2/527?

Though this doesn't apply directly to the patch proposed here but I am wondering in general if we can just change an `EXPORT_SYMBOL_GPL` to `EXPORT_SYMBOL` downstream or would that have any legal implications?

**Comment by Cogitri** [comment]
I'm not a legal expert, but I doubt it'd have legal implications. I just went with this pwtch because other distros (like NixOS) also use it.

**Comment by ncopa** [comment]
Has this been reported and submitted upstream?

What is the upstream's response?

**Comment by Cogitri** [comment]
See: https://github.com/zfsonlinux/zfs/issues/8259

TL;DR upstream "broke" this themselves on purpose and doesn't want to export the symbols for non-GPL compliant stuff anymore.

**Comment by ncopa** [comment]
> Has this been reported and submitted upstream?
> 
> What is the upstream's response?

I read the thread referenced in https://github.com/alpinelinux/aports/pull/8784#issuecomment-501872454

I think we should follow upstream.

If we at some point changes from a `linux-vanilla` to a `linux-alpine` for alpine specific kernel, then we can reconsider this again. But I think we should try avoid adding patches as long as we call the kernel `vanilla`.

**Comment by Cogitri** [comment]
Alrighty
